src/esp/core/managedContainers/ManagedFileBasedContainer.h

```diff
@@ -68,7 +68,7 @@ class ManagedFileBasedContainer : public ManagedContainer<T, Access> {
    */
   ManagedFileIOPtr createObjectFromJSONFile(const std::string& filename,
                                             bool registerObject = true) {
-    io::JsonDocument docConfig = nullptr;
+    std::shared_ptr<io::JsonDocument> docConfig{};
```

Comment:
Why does this one need to be a shared ptr? Pretty sure there is still an openissue to fix this for RapidJSON: https://github.com/Tencent/rapidjson/pull/1916